Server/controllers/authController.js

```diff
@@ -12,22 +12,34 @@ require("dotenv").config();
 const { errorMessages, successMessages } = require("../utils/messages");
 const jwt = require("jsonwebtoken");
 const SERVICE_NAME = "AuthController";
-const { getTokenFromHeaders } = require("../utils/utils");
+const { getTokenFromHeaders, tokenType } = require("../utils/utils");
 const crypto = require("crypto");
 const { handleValidationError, handleError } = require("./controllerUtils");
 
 /**
  * Creates and returns JWT token with an arbitrary payload
  * @function
  * @param {Object} payload
+ * @param {tokenType} typeOfToken - Whether to generate refresh token with long TTL or access token with short TTL.
  * @param {Object} appSettings
  * @returns {String}
  * @throws {Error}
  */
-const issueToken = (payload, appSettings) => {
+const issueToken = (payload, typeOfToken, appSettings) => {
   try {
-    const tokenTTL = appSettings.jwtTTL ? appSettings.jwtTTL : "2h";
-    return jwt.sign(payload, appSettings.jwtSecret, { expiresIn: tokenTTL });
+    let tokenSecret, tokenTTL, payloadData = {};
+
+    if(typeOfToken === tokenType.REFRESH_TOKEN) {
+      tokenSecret = appSettings.refreshTokenSecret;
+      tokenTTL = appSettings.refreshTokenTTL ? appSettings.refreshTokenTTL : "7d";
+    }
+    else {
+      tokenSecret = appSettings.jwtSecret;
+      tokenTTL = appSettings.jwtTTL ? appSettings.jwtTTL : "2h";
+      payloadData = payload;
+    }
+
```

Comment:
Rather than an `if/else`, why don't we build this up as we go?

```
const tokenTTL = tokenType.REFRESH_TOKEN? (appSettings?refreshTokenTTL?? "2h) : (appSettings?.jwtTTL?? "7d");
   ...
   ... // Set other properties
 jwt.sign(payloadData, tokenSecret, {exipresIn: tokenTTL});
```

The fewer levels of indendtation the better usually, especially when we're doing something like building up a request.

Reply:
Replaced.